### ISSUE TITLE
Cannot use UUID as a Field Filter Variable in Postgres

### ISSUE THREAD

**Issue by violetaria** [metabase/metabase#9665]
I’m using Postgres as the database and having trouble using a UUID field as a Field Filter Variable. Our APIs are microservices and as a result, we have each API database separate. So while the UUID field IS a foreign key, it is not in a table that is visible in the database I am trying to query.

### Bugs

- Your browser and the version: Version 73.0.3683.86 (Official Build) (64-bit)
- Your operating system: macOS Mojave
- Your databases: Postgres
- Metabase version: v0.31.2 
- Metabase hosting environment: GKE
- Metabase internal database: Postgres

*Repeatable steps to reproduce the issue*

- write an SQL question
- add a variable
-- variable type = Field Filter
-- field to map = UUID field (not the primary key, but a foreign key to another table in another database)
-- filter widget type = ID
-- required = true
-- no default value
- select and id from the drop down and hit get answer

error:
```
org.postgresql.util.PSQLException: ERROR: operator does not exist: uuid = character varying Hint: No operator matches the given name and argument type(s). You might need to add explicit type casts. Position: 469
```

I've tried setting the Database Model for the UUID field to `Foreign Key` and `Category` but it still returns the same error. I'm not sure what I am missing or if there is a way to do this, but it's making it difficult to create useful dashboards without being able to filter from a drop down.

**Comment by willc417** [comment]
Having the same issue. It looks like this pr was meant to fix it https://github.com/metabase/metabase/pull/8091. Any update here?

**Comment by gszep** [comment]
would be great to finally get this working

**Comment by walterl** [comment]
@willc417 
> Having the same issue. It looks like this pr was meant to fix it #8091. Any update here?

This issue looks like it could be related to #7955. If it is, #12083 may also fix this issue.

**Comment by walterl** [comment]
#12083 has been merged into `master`. Can anyone please check if it fixed this issue too?

**Comment by tanguyqonto** [comment]
I just experienced the exact same issue. Could it be a regression?

**Comment by flamber** [comment]
@tanguyqonto 
This should have been fixed by #12083 included in 0.35.0 (latest is 0.35.3).
Please post "Diagnostic Info" from Admin > Troubleshooting.
And are the steps to reproduce the same as the original post?

**Comment by tanguyqonto** [comment]
@flamber 
My bad I just saw I'm on v0.34.2
Sorry for the noise!
